src/umpire/strategy/DynamicPool.cpp

```diff
@@ -305,6 +320,8 @@ void DynamicPool::mergeFreeBlocks()
 
   using PointerMap = std::map<Pointer, SizeTuple>;
 
+  UMPIRE_REPLAY("() Free blocks before: " << getFreeBlocks());
```

Comment:
@davidbeckingsale, @jdahm, I think we should limit the changes in this PR to just what is required for the fix.  I'd rather not add new replay semantics here.

Reply:
Agree.

Adding asserts or logging would be okay however.

Reply:
Oops, I meant that only to be a log. I'll fix that!